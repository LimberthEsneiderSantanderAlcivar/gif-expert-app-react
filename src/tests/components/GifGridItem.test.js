import {shallow} from 'enzyme'
import GifGridItem from '../../components/GifGridItem';

describe('tests para el componente de', () => {
    const title = 'shingeki No Kyojin';
    const url = 'https://localhost/algomas/algo.coms';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);


    test('debe renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    

    test('debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })


    test('debe de tener una imagen igual al url igual al de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('debe tener la clase animate fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
})
