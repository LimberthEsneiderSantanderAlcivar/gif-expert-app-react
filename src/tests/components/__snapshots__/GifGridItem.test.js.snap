// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests para el componente de debe renderizar el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="shingeki No Kyojin"
    src="https://localhost/algomas/algo.coms"
  />
  <p>
    shingeki No Kyojin
  </p>
</div>
`;
